### PR TITLE
Add contribution economics to guide

### LINKED ISSUE #7190: Add benefits of PR to Contribution guide
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation
The contribution guide could need some more personal motivation as to why opening a PR.

### Feature / Enhancement Description
Add a section that outlines the benefits of a PR vs. custom code solution.

---
Suggestion:

#### Why create a Pull Request?

Buy cheap, buy twice. What? No, this is not the Economics 101 class, but the same is true for Pull Requests.

There are two ways of writing a feature or fixing a bug. Sometimes the quick solution is to just fork or write a Cloud Code function that does what you want. The fallacy only becomes obvious when you think into the future.

Writing a pull request to make the change directly in Parse Server may take you longer than writing some quick custom solution. But it actually takes much less effort over time because of the benefits you get:

- 🚀 Higher efficiency: Your code is examined for efficiency and interoperability with existing features.
- 🛡 Tighter security: Your code is scrutinized for bugs and vulnerabilities by the community and automated security checks to identify any security issues that may arise in the future.
- 🧬 Continuous improvement: If your feature is used by others it is likely to be continuously improved and extended by the community. 
- 💝 Giving back: You give something back to the community that contributed to make the Parse Platform what it is today and for future developers to come.
- 🧑‍🎓 Improve yourself: You learn something new and better understand the inner workings of Parse Server. This will help you to write more efficient code for your own application.

Most importantly, with every pull requests you improve your skills so that in the future it will take you the same or even less time to write your feature directly in Parse Server. Then you get all the benefits above for free — easy choice, right?

If this is your first pull request or you are already an experienced contributor, the Parse Community is there to support you when you write it. Start off by opening a [new issue](#) or [feature suggestion](#), then create your [pull requests](#)! 🚀

### Example Use Case
(None)

### Alternatives / Workarounds
(None)

### 3rd Party References
(None)

### PR DESCRIPTION
**Pull request by mtrezza**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Improving contribution guide to make contributing more appealing.

Related issue: closes #7190

### Approach
- Added contribution economics chapter
- Added vulnerability policy warning
- Improved contribution guide by making the initial chapter more concise
- Added ToC
- Made headlines more concise

### TODOs before merging
(none)

### CODE REVIEW

**Comment by mtrezza** [comment]
@dblythy Your feedback would be appreciated, I noticed you just did a recent PR on the section that I edited.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,29 +1,65 @@
...
 
-When opening an issue, please make sure you follow the templates and provide as much detail as possible.
+> ⚠️ Please do not post a security vulnerability on GitHub or in the Parse Community Forum. Instead, follow the [Parse Community Security Policy](https://github.com/parse-community/parse-server/security/policy).
 
-After you've completed the contribution, you'll need to submit a Pull Request (PR).
+Please completely fill out any templates to provide essential information about your new feature or the bug you discovered.
 
-If you are not familiar with Pull Requests and want to know more about them, you can visit the [Creating a pull request](https://help.github.com/articles/creating-a-pull-request/) article. It contains detailed informations about the process.
+Together we plan out the best conceptual approach for your contribution, so that your and our time is invested in the best possible approach. The discussion often reveals how to leverage existing features of Parse Server to reach your goal with even less effort and in a more sustainable way.
```
Comment (dblythy) [inline CONTRIBUTING.md:28]: Maybe "Together we will plan out"

**Comment by dblythy** [comment]
Looks good @mtrezza, a much needed improvement IMO.

**Comment by mtrezza** [comment]
> Should we perhaps add a line in the "Please do's" around CHANGELOG?

How?

**Comment by dblythy** [comment]
Perhaps:

**Please do's:**
- ...
- Please consider if any changes to the docs are needed or add additional sections in the case of an enhancement or feature.
- Add your feature to the CHANGELOG.

**Comment by mtrezza** [comment]
I wouldn't duplicate info unless it's necessary, we already have this as a checkbox in the PR template.

**Review by davimacedo — APPROVED** [review]
Nice job @mtrezza !

**Comment by mtrezza** [comment]
@dblythy regarding your earlier comment, the revamping of the "Please Do's" into a step-by-step guide could be done in a separate PR so I'm going to merge this.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**